config/jobs/openenclave/jenkins/RhelBuild.Jenkinsfile

```diff
@@ -1,56 +1,79 @@
+        string(name: 'PULL_NUMBER', defaultValue: params.PULL_NUMBER ?:'master',  description: 'Branch/PR to build')
+        string(name: 'BUILD_TYPE', defaultValue: params.BUILD_TYPE ?:'RelWithDebInfo',  description: 'Build Type')
+        string(name: 'LVI_MITIGATION', defaultValue: params.LVI_MITIGATION ?:'ControlFlow',  description: 'LVI Mitigation Strategy')
+        string(name: 'LVI_MITIGATION_SKIP_TESTS', defaultValue: params.LVI_MITIGATION_SKIP_TESTS ?:'OFF',  description: 'Skip LVI_MITIGATION_SKIP_TESTS')
+        string(name: 'USE_SNMALLOC', defaultValue: params.USE_SNMALLOC ?:'ON',  description: 'Use snmalloc for buiild')
+        string(name: 'E2E', defaultValue: params.E2E ?:'OFF',  description: 'End to en set up')
+    }
 
-// Shared library config, check out common.groovy!
-SHARED_LIBRARY="/config/jobs/openenclave/jenkins/common.groovy"
+    environment {
+        // Shared library config, check out common.groovy!
+        SHARED_LIBRARY="/config/jobs/openenclave/jenkins/common.groovy"
+        EXTRA_CMAKE_ARGS="-DLVI_MITIGATION=${params.LVI_MITIGATION} -DLVI_MITIGATION_SKIP_TESTS=${params.LVI_MITIGATION_SKIP_TESTS} -DUSE_SNMALLOC=${params.USE_SNMALLOC}"
+        // Bug with the environment variable, we only need the above string so set to empty
+        LVI_MITIGATION=""
 
-// whether to run as an e2e test
-E2E=env.E2E?env.E2E:"OFF"
+    }
 
-pipeline {
-    options {
-        timeout(time: 180, unit: 'MINUTES') 
+    agent {
+        label "ACC-${LINUX_VERSION}"
     }
-    agent { label "ACC-${LINUX_VERSION}" }
 
     stages {
-        // Check out test infra repo as need shared libs
-        stage('Checkout'){
+        stage('Checkout') {
             steps{
                 cleanWs()
                 checkout scm
             }
         }
 
-        // Temporarily run always as e2e
-        stage('Install Prereqs (Optional)'){
+        // Run CI checks up front, no need to continue if they fail.
+        stage('CI Checks') {
             steps{
                 script{
-                    def runner = load pwd() + "${SHARED_LIBRARY}"
-                    if("${E2E}" == "ON"){
-                        stage("${LINUX_VERSION} Setup"){
-                            try{
-                                runner.cleanup()
-                                runner.checkout("${PULL_NUMBER}")
-                                runner.installOpenEnclavePrereqs()
-                            } catch (Exception e) {
-                                // Do something with the exception 
-                                error "Program failed, please read logs..."
+                    stage("${params.LINUX_VERSION} Build - CI Checks"){
+                        def runner = load pwd() + "${SHARED_LIBRARY}"
+
+                        try{
+                            runner.cleanup()
+                            runner.checkout("${params.PULL_NUMBER}")
+                            runner.checkCI()
+                        } catch (Exception e) {
```

Comment:
We should try and expand this exception list rather than catch anything and fail. I understand that this could come in later so not to hinder your work velocity.